provider/pkg/provider/helm_release.go

```diff
@@ -360,21 +359,12 @@ func (r *helmReleaseProvider) Check(ctx context.Context, req *pulumirpc.CheckReq
 			return nil, err
 		}
 
-		resourceInfo, err := r.computeResourceInfo(new, r.clientSet)
-		if err != nil && errors.Is(err, fs.ErrNotExist) {
-			// Likely because the chart is not readily available (e.g. import of chart where no repo info is stored).
-			// Declare bankruptcy in being able to determine the underlying resources and hope for the best
-			// further down the operations.
-			resourceInfo = nil
-		} else if err != nil {
-			return nil, err
-		}
-
-		if resourceInfo != nil {
-			new.Checksum = resourceInfo.checksum
-			if len(new.ResourceNames) == 0 {
-				new.ResourceNames = resourceInfo.resourceNames
-			}
+		err = r.setComputedInputs(ctx, urn, new)
```

Comment:
It's not obvious to me why `helmReleaseProvider.setComputedInputs` should return the failure message for `"chart"`. Can we change the naming so this makes sense?

It sounds like `setComputedInputs` should be called `setChart` from this error message. Reading the body of `setComputedInputs` makes it seem like this should be called `setTemplate`, but maybe helm thinks a chart and a template are the same. I don't know.

The function definitely should not be called `setComputedInputs`, since Pulumi doesn't have computed inputs.

Reply:
The function's purpose is to compute the `checksum` and `resourceNames` properties by rendering the chart. When it fails, the `chart` parameter is the likely culprit, though technically there could be problems with the `repositoryOpts`. I agree the name isn't perfect but your suggestions aren't better.

Reply:
Ok. I see that `checksum` doesn't appear anywhere in the schema. When a user changes the `chart` property, does `pulumi` show them a diff on the `checksum` property? I'm assuming that the answer is yes and that is what we want. Is that right?

Reply:
It is true that the `checksum` isn't in the schema, because it was not intended for the user to set directly. It does show up in diff. Honestly I don't have a perfect grasp of the nuances, and am seeking opportunities to learn more.

Reply:
I think there is some room for improvement on the UX,[^1] but nothing I can think of is breaking and this fixes a P1. I'm happy here. I'll leave it to @rquitales for the final approval.

[^1]: In the long term, we probably don't want to expose this to the user at all. It sounds like we should change `checksum` to `__checksum` to tell the engine that it isn't user visible, and override `Diff` so that it shows a diff on `chart` (or the appropriate input property) if `new.__checksum != old.__checksum`. To maintain backwards compatibility we will need to listen for `ingnoreChanges=["checksum"]`, but we can do that easily.